src/domain/configure_ftp_cameras.py

```diff
@@ -157,12 +157,14 @@ def accept_and_close(self):
                 i += 1
 
             # Check for cameras old id (prior to modification) and remove them
-            orphan_cameras = [camera for camera in cameras if camera.id not in ui_camera_id]
+            orphan_cameras = (camera for camera in cameras if camera.id not in ui_camera_id and
+                              camera.type == Camera.CameraTypes.FTPCamera)
             for camera in orphan_cameras:
                 cameras.remove(camera)
             for camera in cameras:
                 if camera.id in self.removed_cameras:
-                    cameras.remove(camera)
+                    if camera.type == Camera.CameraTypes.FTPCamera:
+                        cameras.remove(camera)
```

Comment:
This might cause problems, because you're iterating a list ("cameras") and changing it by removing some of its elements. Python might generate an exception when it happens.
The solution is to make a copy of the list in the for:
    for camera in tuple(cameras):